agent/agent.go

```diff
@@ -94,6 +96,52 @@ type outputUnit struct {
 	outputs []*models.RunningOutput
 }
 
+// Print all available plugins
+func PrintPlugins() {
+	fmt.Println("--- Input Plugins ---")
+	for i, _ := range inputs.Inputs {
+		fmt.Print(i + ", ")
+	}
+	fmt.Println("--- Output Plugins ---")
+	for o, _ := range outputs.Outputs {
+		fmt.Print(o + ", ")
+	}
+}
+
+func AddInputPluginToAgent(a *Agent, plugin_name string) {
+	plugin := inputs.Inputs[plugin_name]	
+	input_config := models.InputConfig{
+		Name : plugin_name, 
+		// TODO rest of config?
+	}
+	running_plugin := models.NewRunningInput(plugin(), &input_config)
+	a.Config.Inputs = append(a.Config.Inputs, running_plugin)
+}
+
+func AddOutputPluginToAgent(a *Agent, plugin_name string) {
+	plugin := outputs.Outputs[plugin_name]	
+	output_config := models.OutputConfig{
+		Name : plugin_name, 
+		// TODO rest of config?
+	}
+	running_plugin := models.NewRunningOutput(plugin_name, plugin(), &output_config, 0, 0)
+	a.Config.Outputs = append(a.Config.Outputs, running_plugin)
+}
+
+func PrintAgentInputPlugins(a *Agent) {
```

Comment:
How about a generic ```GetPlugins``` method of an agent that takes in which type of plugins you want?
eg: `func (a *Agent) GetPlugins(type PluginType)`, with PluginType being a wrapper around strings that are either like "input" or "output" or something? They might already have something like this, not sure.